llmfoundry/callbacks/envlogger.py

```diff
@@ -0,0 +1,177 @@
+        self.log_git = log_git
+        self.log_packages = log_packages
+        self.log_nvidia = log_nvidia
+        self.log_docker = log_docker
+        self.log_system = log_system
+        self.log_gpu = log_gpu
+        self.log_distributed = log_distributed
+        self.env_data: dict[str, Any] = {}
+        self.packages_to_log = packages_to_log or self.PACKAGES_TO_LOG
+
+    def _get_git_info(self, repo_path: str) -> dict[str, str]:
+        repo = git.Repo(repo_path)
+        return {
+            'commit_hash': repo.head.commit.hexsha,
+            'branch': repo.active_branch.name,
+        }
+
+    def _get_package_version(self, package_name: str) -> Optional[str]:
+        try:
+            return pkg_resources.get_distribution(package_name).version
+        except pkg_resources.DistributionNotFound:
+            return None
+
+    def _get_system_info(self) -> dict[str, Any]:
+        return {
+            'python_version': platform.python_version(),
+            'os': f'{platform.system()} {platform.release()}',
+            'hostname': socket.gethostname(),
+            'cpu_info': {
+                'model': platform.processor(),
+                'cores': psutil.cpu_count(logical=False),
+                'threads': psutil.cpu_count(logical=True),
+            },
+            'memory': {
+                'total': psutil.virtual_memory().total,
+                'available': psutil.virtual_memory().available,
+            },
+        }
+
+    def _get_gpu_info(self) -> dict[str, Any]:
+        if torch.cuda.is_available():
+            return {
+                'model': torch.cuda.get_device_name(0),
+                'count': torch.cuda.device_count(),
+                'memory': {
+                    'total': torch.cuda.get_device_properties(0).total_memory,
+                    'allocated': torch.cuda.memory_allocated(0),
+                },
+            }
+        return {'available': False}
+
+    def _get_nvidia_info(self) -> dict[str, Any]:
+        if torch.cuda.is_available():
+            nccl_version = torch.cuda.nccl.version()  # type: ignore
+            return {
+                'cuda_version':
+                    torch.version.cuda,  # type: ignore[attr-defined]
+                'cudnn_version': str(torch.backends.cudnn.version()
+                                    ),  # type: ignore[attr-defined]
+                'nccl_version': '.'.join(
+                    map(str, nccl_version),
+                ),
+            }
+
+        return {'available': False}
+
+    def _get_distributed_info(self) -> dict[str, Any]:
+        return {
+            'world_size': int(os.environ.get('WORLD_SIZE', 1)),
+            'local_world_size': int(os.environ.get('LOCAL_WORLD_SIZE', 1)),
+            'rank': int(os.environ.get('RANK', 0)),
+            'local_rank': int(os.environ.get('LOCAL_RANK', 0)),
+        }
+
+    def _get_docker_info(self) -> dict[str, Any]:
+        from mcli import sdk
```

Comment:
Is this not a new dependency?

Reply:
It's not (https://github.com/mosaicml/composer/blob/main/setup.py#L93) made a top level import